### DISCUSSION TITLE
سلسلة الغو

### DISCUSSION THREAD

**Discussion by BENKOUIDERAHMED** [TheAlgorithms/C#1431] [General]
Ex 02:

#include <stdio.h>

int pgcd(int a, int b) {
    // تحقق من صحة المدخلات
   if (a < 0 || b < 0) {
        printf("enter a positives number\n");
        return 1; // إرجاع قيمة غير صحيحة للإشارة إلى خطأ
    }

    // تطبيق خوارزمية أقليدس
    while (b != 0) {
        int temp = a % b;
        a = b;
        b = temp;
    }

    return a;
}

int main() {
    int num1, num2;

    printf("enter first num: ");
    scanf("%d", &num1);

    printf("enter second num: ");
    scanf("%d", &num2);

    int result = pgcd(num1, num2);

    if (result != 1) {
        printf("Gcd  %d and %d are: %d\n", num1, num2, result);
    }

    return 0;
}

**Comment by itsfingerlickinggood** [comment]
20000 stars !. 20000th star by itsfingerlickinggood
